### PR TITLE
[NCL-2101] whitelisted artifacts are not longer printed as graylisted

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

File: cli-wrap/src/main/resources/cli/da-cli-script.py [hunk 1]
```diff
@@ -70,7 +70,6 @@ def helper_print_white_artifacts_products(response, show_artifacts=True):
             else:
                 print("{}:{} {}{}".format(product, product_version, product_support, product_id))
 
-
 class DependencyAnalysis:
 
     def __init__(self):
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli-script.py]: Maybe put this change into solo commit

File: cli-wrap/src/main/resources/cli/da-cli.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ printUsage() {
     echo "    $0 list black"
     echo "        List all artifacts in blacklist."
     echo ""
-    echo "    $0 list white [GROUP_ID:ARTIFACT_ID:VERSION]"
+    echo "    $0 list white [PRODUCT_NAME:VERSION]"
     echo ""
     echo "        Each artifact in the white list is associated to a product."
     echo "        List all artifacts and its associated product in the white list."
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.sh]: Maybe put this change into solo commit (together with the previous one).

Also the message for this option isn't consistent. There is " optionally specify which GAV to show" which is not true.

File: reports-backend/src/main/java/org/jboss/da/listings/impl/dao/ProductVersionDAOImpl.java [hunk 3]
```diff
@@ -83,17 +83,19 @@ public boolean changeProductVersionStatus(String name, String version,
...
     public List<ProductVersion> findProductVersionsWithArtifact(String groupId, String artifactId,
-            String version) {
+            String version, boolean preciseVersion) {
         CriteriaBuilder cb = em.getCriteriaBuilder();
         CriteriaQuery<ProductVersion> cq = cb.createQuery(type);
         Root<ProductVersion> productVersion = cq.from(type);
         Root<WhiteArtifact> artifact = cq.from(WhiteArtifact.class);
         Join<Artifact, GA> ga = artifact.join("ga");
+        if (preciseVersion)
+            version += '%';
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/listings/impl/dao/ProductVersionDAOImpl.java]: The name of the variable is misleading.

**Comment by janinko** [comment]
And also please rebase to master.

**Comment by janinko** [comment]
The first commit isn't relateted to the issue.

File: cli-wrap/src/main/resources/cli/da-cli.sh [hunk 4]
```diff
@@ -23,12 +23,11 @@ printUsage() {
...
     echo "        List all artifacts in blacklist."
     echo ""
-    echo "    $0 list white [GROUP_ID:ARTIFACT_ID:VERSION]"
-    echo ""
+    echo "    $0 list white [PRODUCT_NAME:VERSION]"
     echo "        Each artifact in the white list is associated to a product."
     echo "        List all artifacts and its associated product in the white list."
-    echo "        You can optionally specify which GAV to show, and it will show"
-    echo "        in which product the GAV is associated with"
+    echo "        You can optionaly specify which project to show and it will show"
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.sh]: typo, should be "optionally"

File: cli-wrap/src/main/resources/cli/da-cli.sh [hunk 5]
```diff
@@ -23,12 +23,11 @@ printUsage() {
...
+    echo "        all artifacts in that project."
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.sh]: you show product, not project

**Comment by janinko** [comment]
When you fix the typos and the test passes, you can merge.

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Pull request merged**